aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -150,15 +152,22 @@ void cosh_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+// We manually override tanh because std::tanh does not work with thrust::complex types.
 template<typename scalar_t>
 __host__ __device__ static inline scalar_t tanh_wrapper(scalar_t v) {
   return ::tanh(v);
 }
 
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> tanh_wrapper(thrust::complex<T> v) {
+  return thrust::tanh(v);
+}
+
 void tanh_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "tanh_cuda", [&]() {
-    AT_SKIP_BFLOAT16_IF_NOT_ROCM(scalar_t, "tanh_cuda", [&] {
-      gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "tanh_cuda", [&]() {
```

Comment:
`AT_DISPATCH_FLOATING_AND_C10_COMPLEX_TYPES_AND2`

Reply:
I don't see this macro defined in the master branch. Have added this macro though.